cpp/include/cudf/table/row_operators.cuh

```diff
@@ -138,6 +139,50 @@ __device__ bool equality_compare(Element const lhs, Element const rhs) {
     return lhs == rhs;
 }
 
+/**---------------------------------------------------------------------------*
+ * @brief Computes whether an element in a column is equal to a scalar value
+ *
+ * @tparam Element identifies the type of the element
+ * @tparam has_nulls Indicates the potential for null values in the column
+ *---------------------------------------------------------------------------**/
+template <typename Element, bool has_nulls = true>
+struct compare_with_value {
```

Comment:
This is heavier weight than what we need. @trevorsm7 has a `make_constant_scalar_iterator` utility that will return a constant iterator that returns the value and validity of a `cudf::scalar` in a `thrust::tuple<T,bool>`. With such an iterator, this function object should not be needed.

Reply:
Yes, it looks like permutation iterators would cover this purpose and be more flexible

Reply:
Where would I find make_constant_scalar_iterator?

Not quite sure where permutation iterators would help... unless the constant_scalar_iterator is a permutation iterator?

Reply:
I found the slack conversation discussing the make_constant_scalar_iterator, I guess it isn't in the baseline yet... but I could use the definition from slack.

I'm trying to reconstruct the path of how the code got to this state... and struggling to recall the exact sequence of events.  I suspect it was mostly a matter of what features were ready in different pull requests at different times.

In order to use this make_constant_scalar_iterator approach, I would need to have the scalar in a column.  A previous commit on the original PR had the scalar being loaded into a column.  The current version does not.  I could certainly pull that code back in and I think this would clean up nicely.

Of course, then I don't need to use the scalar_device_view, since I'll be using a column.  So perhaps I should scrap this branch and start over.

Reply:
Removed this from this PR